packages/world/src/WorldProxy.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { StoreCore } from "@latticexyz/store/src/StoreCore.sol";
+import { STORE_VERSION } from "@latticexyz/store/src/version.sol";
+import { IStoreEvents } from "@latticexyz/store/src/IStoreEvents.sol";
+import { WORLD_VERSION } from "./version.sol";
+import { IWorldEvents } from "./IWorldEvents.sol";
+import { AccessControl } from "./AccessControl.sol";
+import { ROOT_NAMESPACE_ID } from "./constants.sol";
+import { Proxy } from "./Proxy.sol";
+import { StorageSlot } from "./StorageSlot.sol";
+
+/**
+ * @dev Storage slot with the address of the current implementation.
+ * This is the keccak-256 hash of "eip1967.proxy.implementation" subtracted by 1.
+ */
+// solhint-disable-next-line private-vars-leading-underscore
+bytes32 constant IMPLEMENTATION_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+
+/**
+ * @dev Stores a new address in the EIP1967 implementation slot.
+ */
+function _setImplementation(address newImplementation) {
+  StorageSlot.getAddressSlot(IMPLEMENTATION_SLOT).value = newImplementation;
+}
```

Comment:
Is there some standard around whether to emit an event when the implementation changed?

Reply:
There is, but I think we should move away from the standards altogether (ie. not referring to this as EIP1967)